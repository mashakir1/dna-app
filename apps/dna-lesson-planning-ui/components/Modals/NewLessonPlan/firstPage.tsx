// Imports
import { Dispatch, SetStateAction } from "react";
import useTranslation from "next-translate/useTranslation";

// Components
import Spacing from "components/Layout/Spacing";
import { BaseCard } from "components/Cards";
import { LearnerAmbitionStar, OpenBookIcon } from "components/Svg";
import { IconWithBackground } from "components/Icons";

// Styles
import styles from "./newLessonPlan.module.scss";

interface FirstPageCardProps {
  setPageNumber: Dispatch<SetStateAction<number>>;
  icon: JSX.Element;
  title: string;
  subtitle: string;
}

const FirstPageCard: (props: FirstPageCardProps) => JSX.Element = ({
  setPageNumber,
  icon,
  title,
  subtitle
}) => (
  <BaseCard
    bordered
    classNames={`${styles.lessonPlanCard} flex column items-center`}
    onClick={() => setPageNumber(1)}
  >
    {/* TODO - implement cascading hover state to Icon and IconBackground colour */}
    <IconWithBackground icon={icon} classNames={styles.firstPageIconWithBackground} />
    <Spacing height={"m"} />
    <div className={`${styles["lessonPlanCard-titles"]}`}>
      <p className={"large"}><b>{title}</b></p>
      <Spacing height={"xs"} />
      <p className={"small"}>{subtitle}</p>
    </div>
  </BaseCard>
);

interface FirstPageProps {
  setPageNumber: Dispatch<SetStateAction<number>>;
}

export const NewLessonPlanFirstPage: (props: FirstPageProps) => JSX.Element = ({ setPageNumber }) => {
  const { t } = useTranslation("common");

  return (
    <div className={`${styles.newLessonPlan} flex column items-center`}>
      <h6>{t("planning-choice")}</h6>
      <Spacing height={"xxl"} />
      <div className={styles.cardWrapper}>
        <FirstPageCard
          setPageNumber={setPageNumber}
          icon={<OpenBookIcon className={styles.firstPageIcon} />}
          title={t("plan-by-subject-title")}
          subtitle={t("plan-by-subject-subtitle")}
        />
        <FirstPageCard
          setPageNumber={setPageNumber}
          icon={<LearnerAmbitionStar className={styles.firstPageIcon} />}
          title={t("plan-by-learner-ambition-title")}
          subtitle={t("plan-by-learner-ambition-subtitle")}
        />
      </div>
    </div>
  );
};
