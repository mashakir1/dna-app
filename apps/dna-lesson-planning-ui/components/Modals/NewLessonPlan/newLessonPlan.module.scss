.newLessonPlan {
  width: 100%;
}

.cardWrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  gap: $spacing_m;
  width: 100%;
}

.lessonPlanCard {
  padding: $spacing-l $spacing-m;
  width: 100%;

  &-titles {
    max-width: 190px;
    text-align: center;

    > * {
      margin: 5px 0;
    }
  }

  &:hover {
    background-color: $colour_background_teal;
    cursor: pointer;

    .firstPageIcon path {
      fill: $colour_ui_teal;
    }

    .firstPageIconWithBackground {
      background-color: $colour_light_teal;
    }
  }
}
