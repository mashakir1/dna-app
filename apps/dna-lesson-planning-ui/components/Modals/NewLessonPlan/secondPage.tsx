// Imports
import { useState } from "react";
import useTranslation from "next-translate/useTranslation";

// Components
import Spacing from "components/Layout/Spacing";
import { FilledLinkButton } from "components/Buttons";
import SelectWrapper from "./selectWrapper";

// Styles
import styles from "./newLessonPlan.module.scss";

export const NewLessonPlanSecondPage: () => JSX.Element = () => {
  const { t } = useTranslation("common");

  const [yearSelection, setYearSelection] = useState<string>(undefined);
  const [subjectSelection, setSubjectSelection] = useState<string>(undefined);

  // TODO - save selection values in session
  return (
    <div className={`${styles.newLessonPlan} flex column justify-center items-center`}>
      <h6>{t("year-and-subject-choice-title")}</h6>
      <Spacing height={"m"} />
      <SelectWrapper
        yearSelection={yearSelection}
        setYearSelection={setYearSelection}
        setSubjectSelection={setSubjectSelection}
      />
      <Spacing height={"m"} />
      <FilledLinkButton
        text={"Start Planning"}
        href={"/plan/subject"}
        disabled={!subjectSelection}
      />
    </div>
  );
};
