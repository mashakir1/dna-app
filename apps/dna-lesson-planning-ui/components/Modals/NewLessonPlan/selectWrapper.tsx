// Imports
import { Dispatch, SetStateAction } from "react";
import useTranslation from "next-translate/useTranslation";

// Components
import { Select } from "components/Forms/Select";
import Spacing from "components/Layout/Spacing";

interface WrapperProps {
  yearSelection: string;
  setYearSelection: Dispatch<SetStateAction<string>>;
  setSubjectSelection: Dispatch<SetStateAction<string>>;
}

const SelectWrapper: (props: WrapperProps) => JSX.Element = ({
  yearSelection,
  setYearSelection,
  setSubjectSelection
}) => {
  const { t } = useTranslation("common");

  // TODO pull options from backend
  const yearOptions = [{ label: "Year 3", value: "year-3" }];
  const subjectOptions = [{ label: "Science", value: "science" }];

  return <>
    <Select
      key={"YearSelect"}
      name={"YearSelect"}
      placeholder={t("select-year-placeholder")}
      options={yearOptions}
      onChange={setYearSelection}
    />
    <Spacing height={"s"} />
    <Select
      key={"SubjectSelect"}
      name={"SubjectSelect"}
      placeholder={t("select-subject-placeholder")}
      options={subjectOptions}
      onChange={setSubjectSelection}
      isDisabled={!yearSelection}
    />
  </>;
};

export default SelectWrapper;
