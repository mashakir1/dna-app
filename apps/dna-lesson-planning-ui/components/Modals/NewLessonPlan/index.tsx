// Imports
import { Dispatch, SetStateAction, useState } from "react";

// Components
import { Modal } from "..";
import { NewLessonPlanFirstPage } from "./firstPage";
import { NewLessonPlanSecondPage } from "./secondPage";

interface NewLessonPlanProps {
  visible: boolean;
  setVisible: Dispatch<SetStateAction<boolean>>;
}

const NewLessonPlan: (props: NewLessonPlanProps) => JSX.Element = ({
  visible,
  setVisible
}) => {
  const [pageNumber, setPageNumber] = useState<number>(0);

  return <Modal
    visible={visible}
    closeButton
    setVisible={setVisible}
    paginationProps={{ pageNumber, setPageNumber }}
  >
    {pageNumber === 0
      ? <NewLessonPlanFirstPage setPageNumber={setPageNumber} />
      : <NewLessonPlanSecondPage />}
  </Modal>;
};

export default NewLessonPlan;
