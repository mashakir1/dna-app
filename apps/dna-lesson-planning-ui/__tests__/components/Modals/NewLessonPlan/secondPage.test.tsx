import { fireEvent, render, screen } from "@testing-library/react";
import { NewLessonPlanSecondPage } from "components/Modals/NewLessonPlan/secondPage";

jest.mock("next/link", () => {
  return ({ children }) => {
    return children;
  };
});

describe("secondPage", () => {
  it("matches snapshot", () => {
    const secondPage = render(<NewLessonPlanSecondPage />).container;

    expect(secondPage).toMatchSnapshot();
  });

  it("is not submittable without filling in form", () => {
    render(<NewLessonPlanSecondPage />);

    const pageBeforeClick = window.location.pathname;
    const startPlanningButton = screen.getByText("Start Planning");
    fireEvent.click(startPlanningButton);
    const pageAfterClick = window.location.pathname;

    expect(pageBeforeClick).toEqual(pageAfterClick);
  });

  it("is submittable with filled form", () => {
    // TODO
  });
});
