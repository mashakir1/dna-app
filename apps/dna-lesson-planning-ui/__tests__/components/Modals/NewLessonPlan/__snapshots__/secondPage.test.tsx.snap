// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`secondPage matches snapshot 1`] = `
<div>
  <div
    class="newLessonPlan flex column justify-center items-center"
  >
    <h6>
      year-and-subject-choice-title
    </h6>
    <div
      class="spacingM"
    />
    <div
      class=" css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class="reactSelect__control css-olrejj-control"
      >
        <div
          class="reactSelect__value-container css-kqoj1a-ValueContainer"
        >
          <div
            class="reactSelect__placeholder css-spmjzn-placeholder"
            id="react-select-2-placeholder"
          >
            select-year-placeholder
          </div>
          <div
            class="reactSelect__input-container css-1igp90l-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class="reactSelect__input"
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="reactSelect__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.29289 8.29289C5.68342 7.90237 6.31658 7.90237 6.70711 8.29289L12 13.5858L17.2929 8.29289C17.6834 7.90237 18.3166 7.90237 18.7071 8.29289C19.0976 8.68342 19.0976 9.31658 18.7071 9.70711L12.7071 15.7071C12.3166 16.0976 11.6834 16.0976 11.2929 15.7071L5.29289 9.70711C4.90237 9.31658 4.90237 8.68342 5.29289 8.29289Z"
              fill="#30CCD6"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <input
        name="YearSelect"
        type="hidden"
        value=""
      />
    </div>
    <div
      class="spacingS"
    />
    <div
      class="reactSelect--is-disabled css-3iigni-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-3-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class="reactSelect__control reactSelect__control--is-disabled css-6rs54r-control"
      >
        <div
          class="reactSelect__value-container css-kqoj1a-ValueContainer"
        >
          <div
            class="reactSelect__placeholder css-19zpqpo-placeholder"
            id="react-select-3-placeholder"
          >
            select-subject-placeholder
          </div>
          <div
            class="reactSelect__input-container css-izxaji-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-describedby="react-select-3-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class="reactSelect__input"
              disabled=""
              id="react-select-3-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="reactSelect__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.29289 8.29289C5.68342 7.90237 6.31658 7.90237 6.70711 8.29289L12 13.5858L17.2929 8.29289C17.6834 7.90237 18.3166 7.90237 18.7071 8.29289C19.0976 8.68342 19.0976 9.31658 18.7071 9.70711L12.7071 15.7071C12.3166 16.0976 11.6834 16.0976 11.2929 15.7071L5.29289 9.70711C4.90237 9.31658 4.90237 8.68342 5.29289 8.29289Z"
              fill="#30CCD6"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="spacingM"
    />
    <span>
      Start Planning
    </span>
  </div>
</div>
`;
